docs/DeveloperGuide.md

```diff
@@ -1,38 +1,164 @@
+<!-- TOC -->
+
+
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+###### [Back to table of contents](#table-of-contents)
+
+## Product Scope
+
+### Target User Profile
+* People who go to gym and do exercise
+* People who want to lose weight or strength muscles
+
+### Value Proposition
+Help our target users to record down the exercise, and they can check the exercise whenever they want.
+Besides, this fitness app also help them to calculate the calories consumption for each workout.
+
 
-## Design & implementation
+###### [Back to table of contents](#table-of-contents)
 
+## Design & Implementation
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Architecture
+
+###### [Back to table of contents](#table-of-contents)
+
+We aim to help fellow gym enthusiasts and new gym members to help keep
+track of their future and present training.
+=======
+### UI component
+
+###### [Back to table of contents](#table-of-contents)
+
+### Start component
+
+###### [Back to table of contents](#table-of-contents)
+
+### Add component
+
+The add mechanism is facilitated by `AddCommand`. 
+It extends `Command` and modifies the execute function to add an exercise.
+
+<img src="images/AddExerciseDiagram.png" width="450" />
+
+Given below is an example usage scenario and how the add mechanism behaves at each step.
+
+Step 1. The user enters the add command with the necessary arguments.
+
+Step 2. The input is processed by the `Parser` to separate out the arguments and creates the exercise to be added, 
+`toAdd`.
+
+Step 3. The `AddCommand` calls `execute()` which calls `WorkList#getCurrentWorkout()` to return `currentWorkout`.
+
+Step 4. Finally, `addExercise()` is called and `toAdd` is added to `currentWorkout`.
+
+###### [Back to table of contents](#table-of-contents)
+
+### List component
 
-## Product scope
-### Target user profile
+###### [Back to table of contents](#table-of-contents)
 
-{Describe the target user profile}
+### View component
+The View component is facilitated by `Parser`,`Ui`,`WorkoutList`,`Command` and `ViewCommand`, where the user will 
+enter a specific workout date and the number of exercises on that date will be displayed
 
-### Value proposition
+Below are the specific steps on how to use the view function and how the mechanism will flow:
+
+* Step 1: We will assume that the user has started a workout on two specific dates, `11/02/22` which was added with the following command `/start 11/02/22`
+  and `12/02/22`, which was added with the following command `/start 12/02/22`.
+* Step 2: The user will then use the following command `/view 11/02/22` will be taken into the parser
+  and will return a list of exercises done on that specified date.
+
+<img src="images/ViewDiagram.png" width="450" />
```

Comment:
Activation bar at the bottom not closed for both `Duke` and `Parser` entities.
Missing colons ( : ) for the entities `Parser`, `CheckInputs`, `ViewWorkoutCommand`. Require fix.